apps/api/middleware/checkToken.ts

```diff
@@ -0,0 +1,30 @@
+import admin from 'firebase-admin';
+
+import serviceAccount from '../service-account-key.json';
+
+admin.initializeApp({
+	credential: admin.credential.cert({
+		projectId: serviceAccount.project_id,
+		clientEmail: serviceAccount.client_email,
+		privateKey: serviceAccount.private_key,
+	}),
+});
```

Comment:
put `service-account-key.json` content under `.env`, e.g:
```bash
FIREBASE_PROJECT_ID="..."
FIREBASE_CLIENT_EMAIL="..."
FIREBASE_PRIVATE_KEY="..."
```

and then, use it like following:
```typescript
admin.initializeApp({
	credential: admin.credential.cert({
		projectId: process.env.FIREBASE_PROJECT_ID,
		clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
		privateKey: process.env.FIREBASE_PRIVATE_KEY,
	}),
});
```

or event leaner with config object:
```typescript
// utils/config.ts
export const environments = {
  firebaseProjectId: process.env.FIREBASE_PROJECT_ID,
	firebaseClientEmail: process.env.FIREBASE_CLIENT_EMAIL,
	firebasePrivateKey: process.env.FIREBASE_PRIVATE_KEY,
};

admin.initializeApp({
	credential: admin.credential.cert({
		projectId: environments.firebaseProjectId, // with code completion
		clientEmail: environments.firebaseClientEmail,
		privateKey: environments.firebasePrivateKey,
	}),
});
```